From b55a7edef50d5de3bd9ef8d13b6e2abaa1e73eec Mon Sep 17 00:00:00 2001
From: "shi.liu" <shi.liu@amlogic.com>
Date: Wed, 30 Jul 2014 17:46:50 +0800
Subject: [PATCH 4601/5965] PD #93752: mm: disable free_unused_memmap() on
 mem_init

alloc_node_mem_map() allocates pages struct whoes number is the same as
node_spanned_pages.
This allocation contains whole pages of all memory (reserved memory contained).
To save memory and reserved memory not contained in memory management
subsystem. The mem_init() frees whole pages struct of reserved memory.
Meanwhile if the page struct(such as page->flags) of reserved memory was modified
by some drivers, memory trample occurs;

This commit disable free_unused_memmap() on mem_init, several megabytes is wasted.
---
 arch/arm/mm/init.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
index 0ecc43fd6229..88b349d254d6 100644
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -594,8 +594,9 @@ void __init mem_init(void)
 	max_mapnr   = pfn_to_page(max_pfn + PHYS_PFN_OFFSET) - mem_map;
 
 	/* this will put all unused low memory onto the freelists */
+#ifndef CONFIG_PLAT_MESON
 	free_unused_memmap(&meminfo);
-
+#endif
 	totalram_pages += free_all_bootmem();
 
 #ifdef CONFIG_SA1111
-- 
2.19.0

